### PR TITLE
Omitted SAVANI test prm file and Limiting the max order for tests

### PR DESCRIPTION
**Pull request by Shangxin-Liu**
When I performed tests on my new built ASPECT from public version, I just realized that I forgot attaching the test prm file for SAVANI in my last year's pull request. I have checked again carefully. The screen and solution output stuff is already on the repository and the test prm file is the only omitted one. To make sure, I performed SAVANI test again on my cluster:

[shangxin@brlogin2 build]$ ctest -R SAVANI -V
UpdateCTestConfiguration  from :/home/shangxin/aspect/build/DartConfiguration.tcl
 Add coverage exclude regular expressions.
UpdateCTestConfiguration  from :/home/shangxin/aspect/build/DartConfiguration.tcl
Test project /home/shangxin/aspect/build
Constructing a list of tests
Done constructing a list of tests
Checking test dependency graph...
Checking test dependency graph end
test 123
    Start 123: initial_condition_SAVANI

123: Test command: /opt/apps/cmake/2.8.11.2/bin/cmake "-DBINARY_DIR=/home/shangxin/aspect/build" "-DTESTNAME=tests.initial_condition_SAVANI" "-DERROR="Test initial_condition_SAVANI failed"" "-P" "/home/shangxin/aspect/tests/run_test.cmake"
123: Test timeout computed to be: 600
123: tests.initial_condition_SAVANI: success.
1/1 Test #123: initial_condition_SAVANI .........   Passed   13.55 sec

The following tests passed:
	initial_condition_SAVANI

100% tests passed, 0 tests failed out of 1

Sorry for my omission....

### CODE REVIEW

**Comment by tjhei** [comment]
close but not quite, @bangerth /run-tests :-)

**Comment by Shangxin-Liu** [comment]
I've run the test on my cluster and it just output exactly the same screen-output, .gnuplot, and statistic files as in the repository. I didn't find a way to attach the duplicate files so I just copied the test result here. I think this minor difference is due to our different clusters. Maybe you could attach the output files from your cluster@tjhei?

**Comment by tjhei** [comment]
just apply this patch: http://www.math.clemson.edu/~heister/aspect-logs/cc298d0c9ef584768799383ab26ecb1f7f4f260c/changes.diff

**Comment by Shangxin-Liu** [comment]
There are some "trailing whitespace" warnings when I run git apply on this patch but the files indeed change. Does it matter?

**Comment by Shangxin-Liu** [comment]
Oops... It seems this warning matters....

**Comment by bangerth** [comment]
The test times out on the tester after 10 minutes. @Shangxin-Liu -- how long does it run on your machine?

**Comment by Shangxin-Liu** [comment]
I run it in Release mode, it's faster @bangerth 

Number of active cells: 96 (on 1 levels)
Number of degrees of freedom: 4,030 (2,910+150+970)

*** Timestep 0:  t=0 seconds
   Solving temperature system... 0 iterations.
   Rebuilding Stokes preconditioner...
   Solving Stokes system... 30+16 iterations.

   Postprocessing:
     Writing graphical output: output-initial_condition_SAVANI/solution-00000
     Temperature min/avg/max:  1484 K, 1599 K, 1783 K

Termination requested by criterion: end time


+---------------------------------------------+------------+------------+
| Total wallclock time elapsed since start    |       236s |            |
|                                             |            |            |
| Section                         | no. calls |  wall time | % of total |
+---------------------------------+-----------+------------+------------+
| Assemble Stokes system          |         1 |     0.274s |      0.12% |
| Assemble temperature system     |         1 |    0.0977s |     0.041% |
| Build Stokes preconditioner     |         1 |      0.16s |     0.068% |
| Build temperature preconditioner|         1 |    0.0306s |     0.013% |
| Solve Stokes system             |         1 |     0.243s |       0.1% |
| Solve temperature system        |         1 |   0.00116s |   0.00049% |
| Initialization                  |         2 |       235s |     1e+02% |
| Postprocessing                  |         1 |    0.0553s |     0.023% |
| Setup dof systems               |         1 |    0.0948s |      0.04% |
+---------------------------------+-----------+------------+------------+

**Comment by Shangxin-Liu** [comment]
If so, maybe in the test, we can set the max order to 20 or 30 instead of 60?

**Comment by Shangxin-Liu** [comment]
It seems that gcc is fine here but clang times out..

**Comment by bangerth** [comment]
Yes, I think that's what we need to do. The tester runs the tests in debug mode. If it takes 236s in release mode, it would take around 2000 seconds in debug mode. Can you try how long it takes if you reduce the order to 20? The point of the test is not that it must be completely realistic, but so that we can see if a future change breaks existing functionality. This can equally well be checked with an order of 20.

**Comment by Shangxin-Liu** [comment]
Ok. Now the SAVANI code is written to read the max order from the first line of the data file (60). I can create another input parameter called "max order" to control this. But if so I suppose we need to also add this parameter to S40RTS to make them consistent. Another option is to create another data input file called SAVANI_test.ab or something else including only the first 20 order data (just like S20RTS data).

I think maybe the first option is better because users can choose any max order value they want.

What do you think of this? @bangerth @tjhei

File: tests/initial_condition_SAVANI.prm [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    set Outer temperature = 1600
+  end
+end
+
+
+subsection Initial conditions
+  set Model name = SAVANI perturbation
+  subsection SAVANI perturbation
+    set Initial condition file name           = savani.dlnvs.60.m.ab
+    set vs to density scaling                 = 0.3
```
Comment (bangerth) [inline tests/initial_condition_SAVANI.prm]: This is unrelated to the current patch, and may be addressed separately: Could we rename this parameter so that "Vs" is capitalized? We always do that with parameters, and in this case I think there isn't even much of an argument that it might be a symbol because people will probably have an easier time reading Vs over vs when thinking of the shear wave velocity.

Comment (bangerth) [inline tests/initial_condition_SAVANI.prm]: Ah, I see that unfortunately, the S40RTS model has the same problem.

Let's leave it as is then.

Comment (bangerth) [inline tests/initial_condition_SAVANI.prm]: See #760

**Comment by bangerth** [comment]
@Shangxin-Liu -- did you forget to push that last commit? I can't see any change to the code that reads or limits the order.

I would favor a system whereby the data file (e.g., savani.dlnvs.60.m.ab) declares the order of the spherical harmonic expansion it stores (i.e., make the data file *descriptive*) but where a user can specify *in the input file* a lower order. If the order in the input file is smaller than the one in the data file, then the code will simply discard any higher modes.

**Comment by Shangxin-Liu** [comment]
I haven't changed the code to add limiting order option. Now the max order of S40RTS/SAVANI is fixed to 20 or 40 or 60 from the data files. I can do this later and make separate pull request with also the upper case Vs. @bangerth

**Comment by Shangxin-Liu** [comment]
But as for the test output files, I suppose we may need to use the ones from Timo's cluster. @bangerth

**Comment by bangerth** [comment]
But then I still don't see how we can run the test within the time limit. We need to do something about limiting the order of the expansion before we can merge the test...

**Comment by Shangxin-Liu** [comment]
Yes. I just changed the name of this pull request. I will make another commit later on this. @bangerth

**Comment by Shangxin-Liu** [comment]
Finished adding limiting max order in S40RTS and SAVANI. The SAVANI test is limited to use up to 40 order. All tests results are attached but maybe slightly different from Timo's cluster. @bangerth @tjhei

**Comment by gassmoeller** [comment]
Just checking this PR it seems this would be in principle ready to merge, except that the results on Timo's tester are slightly different from yours @Shangxin-Liu is that correct?

**Comment by Shangxin-Liu** [comment]
Yes. The only slight misfit is due to cluster difference. @gassmoeller

**Comment by gassmoeller** [comment]
Ok, great. @tjhei: When you have a minute could you update the test results? Then we can finally merge.

**Comment by gassmoeller** [comment]
Merged, thanks for the fix :+1:
